version: "3.8"

services:
  api-service:
    build:
      dockerfile: Dockerfile
      context: ./src/be
      target: prod
    container_name: api-service
    hostname: api-service
    restart: on-failure
    ports:
      - "7780:7780"
    networks:
      stima-network:
        aliases:
          - api.local
    environment:
      DATABASE_HOST: postgresql.stima.local
      DATABASE_PORT: 5432
  fe-service:
    build:
      dockerfile: Dockerfile
      context: ./src/fe
      target: prod
    container_name: fe-service
    hostname: fe-service
    restart: on-failure
    ports:
      - "7781:3000"
    networks:
      stima-network:
        aliases:
          - fe.local
  postgresql:
    image: postgres:latest
    container_name: postgresql
    hostname: postgresql
    restart: always
    environment:
      POSTGRES_USER: postgres
      POSTGRES_PASSWORD: postgres
      POSTGRES_DB: auth-db
    volumes:
      - ./src/postgresql/data:/var/lib/postgresql/data
      - ./src/postgresql/init/init.sql:/docker-entrypoint-initdb.d/init.sql
    ports:
      - 9001:5432
    networks:
      stima-network:
        aliases:
          - postgresql.stima.local
    command: ["postgres", "-c", "wal_level=logical"]

networks:
  stima-network:
